esphome/components/ili9xxx/ili9xxx_display.cpp

```diff
@@ -162,86 +170,80 @@ void ILI9XXXDisplay::update() {
 }
 
 void ILI9XXXDisplay::display_() {
-  // we will only update the changed window to the display
-  uint16_t w = this->x_high_ - this->x_low_ + 1;  // NOLINT
-  uint16_t h = this->y_high_ - this->y_low_ + 1;  // NOLINT
-  uint32_t start_pos = ((this->y_low_ * this->width_) + x_low_);
-
+  uint8_t transfer_buffer[ILI9XXX_TRANSFER_BUFFER_SIZE];
   // check if something was displayed
   if ((this->x_high_ < this->x_low_) || (this->y_high_ < this->y_low_)) {
     ESP_LOGV(TAG, "Nothing to display");
     return;
   }
 
-  set_addr_window_(this->x_low_, this->y_low_, w, h);
+  // we will only update the changed rows to the display
+  uint16_t const w = this->x_high_ - this->x_low_ + 1;  // NOLINT
+  uint16_t const h = this->y_high_ - this->y_low_ + 1;  // NOLINT
+  size_t pos = (this->y_low_ * this->width_) + this->x_low_;
 
   ESP_LOGV(TAG,
            "Start display(xlow:%d, ylow:%d, xhigh:%d, yhigh:%d, width:%d, "
-           "heigth:%d, start_pos:%d)",
-           this->x_low_, this->y_low_, this->x_high_, this->y_high_, w, h, start_pos);
-
-  this->start_data_();
-  for (uint16_t row = 0; row < h; row++) {
-    uint32_t pos = start_pos + (row * width_);
-    uint32_t rem = w;
-
-    while (rem > 0) {
-      uint32_t sz = std::min(rem, ILI9XXX_TRANSFER_BUFFER_SIZE);
-      // ESP_LOGVV(TAG, "Send to display(pos:%d, rem:%d, zs:%d)", pos, rem, sz);
-      buffer_to_transfer_(pos, sz);
+           "height:%d, mode=%d, 18bit=%d)",
+           this->x_low_, this->y_low_, this->x_high_, this->y_high_, w, h, this->buffer_color_mode_,
+           this->is_18bitdisplay_);
+
+  set_addr_window_(this->x_low_, this->y_low_, this->x_high_, this->y_high_);
+  if (this->buffer_color_mode_ == BITS_16 && !this->is_18bitdisplay_) {
+    // 16 bit mode maps directly to display format
+    while  (rem-- != 0) {
```

Comment:
Oeps my mistake to hasty with copy paste here

```suggestion
    while  (rem != 0) {
       App.feed_wdt();
```